### PR TITLE
Change delete index API to logical delete

### PR DESCRIPTION
**Pull request by dai-chen**
### Description

This is a follow-up PR after https://github.com/opensearch-project/opensearch-spark/pull/189 which changes the semantic of delete index API from physical to logical delete.

### Issues Resolved

https://github.com/opensearch-project/opensearch-spark/issues/104

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by penghuo — APPROVED** [review]
no doc change required?

**Comment by dai-chen** [comment]
> no doc change required?

Missed it. Updated user manual here: https://github.com/dai-chen/opensearch-spark/blob/change-logical-delete/docs/index.md#sql. Thanks!

File: docs/index.md [hunk 1]
```diff
@@ -122,6 +122,13 @@ High level API is dependent on query engine implementation. Please see Query Eng
 
 ### SQL
 
+- **CREATE:** Create a Flint index with the specified indexing logic. If the auto-refresh option is enabled, a background job will continually update the index with the latest data from the source.
+- **REFRESH:** Manually refresh a Flint index. This command is applicable only to indexes with the auto-refresh option disabled.
+- **SHOW:** Display all Flint indexes in the specified catalog or source table.
+- **DESCRIBE:** Retrieve detailed information about a Flint index.
+- **DROP:** Delete a Flint index logically. This action stops the refreshing process and rules it out in query rewrite.
+- **VACUUM:** Physically remove all data associated with a Flint index, including index metadata and data. This operation effectively cleans up and frees resources.
```
Comment (vamsi-amazon) [inline docs/index.md:130]: Just curious, is there any precedence for this work "VACCUM"

Comment (dai-chen) [reply docs/index.md:130]: you can find more details here: https://github.com/opensearch-project/opensearch-spark/pull/189

**Pull request merged**